Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -290,6 +290,72 @@ For white-box source code testing, you'll need a setup similar to the developer'
+
+###### Native Code
+
+In order to extract strings from native code used in an Android application, you can use GUI tools such as Ghidra or Cutter or rely on CLI-based tools such as the _strings_ Unix utility (`strings <path_to_binary>`) or radare2's rabin2 (`rabin2 -zz <path_to_binary>`). When using the CLI-based ones you can take advantage of other tools such as grep (e.g. in conjunction with regular expressions) to further filter and analyze the results. 
+
+##### Cross References
+
+###### Java and Kotlin
+
+Java cross references is supported by many tools, including Ghidra and [jadx](https://github.com/skylot/jadx "jadx"). With Ghidra cross references can be obtained by right clicking on the desired function in the symbol tree and selecting **Show References to**, as shown in the image below.
+
+<img src="Images/Chapters/0x05c/Ghidra_java_xrefs.png" width="500px"/>
+
+<img src="Images/Chapters/0x05c/jadx_java_xrefs.png" width="500px"/>
+
+###### Native Code
+
+Similarly to Java analysis, you can also use Ghidra to analyze native libraries and obtain cross references by right clicking the desired function and selecting **Show References to**.
+
+<img src="Images/Chapters/0x05c/ghidra_native_xrefs.png" width="500px"/>
+
+##### API Usage
+
+The Android platform provides many in-built libraries for frequently used functionalities in applications, for example cryptography, Bluetooth, NFC, network or location libraries. Determining the presence of these libraries in an application can give us valuable information about its nature.
+
+For instance, if an application is importing `javax.crypto.Cipher`, it indicates that the application will be performing some kind of cryptographic operation. Fortunately, cryptographic calls are very standard in nature, i.e, they need to be called in a particular order to work correctly, this knowledge can be helpful when analyzing cryptography APIs. For example, by looking for the `Cipher.getInstance` function, we can determine the cryptographic algorithm being used. With such an approach we can directly move to analyzing cryptographic assets, which often are very critical in an application. Further information on how to analyse Android's cryptographic APIs is discussed in the section  "[Android Cryptographic APIs](0x05e-testing-cryptography "Android Cryptographic APIs")".
+
+Similarly, the above approach can be used to determine where and how an application is using NFC. For instance, an application using Host-based Card Emulation for performing digital payments must use the `android.nfc` package. Therefore, a good stating point for NFC API analysis would be to consult the [Android Developer Documentation](https://developer.android.com/guide/topics/connectivity/nfc/hce "Host-based card emulation overview") to get some ideas and start searching for critical functions such as `processCommandApdu` from the `android.nfc.cardemulation.HostApduService` class. 
+
+
+##### Checking Secure Connections
```

Comment:
Maybe we can generalize and just call it like this. This way we remain more flexible. One thing to add (after this PR is merged) is something like #1707 (Domain enumeration).

For the things mentioned by Sven above (https://github.com/OWASP/owasp-mstg/pull/1686/files#r408075578) simply refer to "Basic Network Monitoring/Sniffing" in 0x05b. And for the actual test cases refer to 0x05g.

```suggestion
##### Network Communication
```